### ISSUE TITLE
Decompilation result seems to be cached until language server workspace is cleaned

### ISSUE THREAD

**Issue by sbreakey** [redhat-developer/vscode-java#3244]
When you click on a class file and then click "Decompile class file", the results of this seem to be cached, even if that class file is regenerated.
It appears the only way to update that cache is to clean the workspace, and then decompile again.
If this is indeed the case, could the modified date of the .class file not be used to invalidate the cache?

##### Environment
Operating System: MacOS 12.6.7
JDK version: openjdk version "17.0.8" 2023-07-18
OpenJDK Runtime Environment Homebrew (build 17.0.8+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.8+0, mixed mode, sharing)
Visual Studio Code version: 1.81.1
Java extension version: 1.21.0

**Comment by testforstephen** [comment]
Yes, we implemented a cache for the decompiled class file. However, reloading the window will invalidate the cache.

It makes sense to add some policy to invalidate it if the class file has been modified.
